source/Clipboard.js

```diff
@@ -83,7 +83,26 @@ var onPaste = function ( event ) {
     if ( items && ( hasFiles || ( !isEdge && !hasHtml ))) {
         event.preventDefault();
         l = items.length;
-        while ( l-- ) {
+
+        // Trigger a willPaste event if these is an image type on the clipboardData.
+        for (var i = items.length - 1; i >= 0; i--) {
+            if ( /^image\/.*/.test( items[i].type ) ) {
+                hasImage = true;
+                var imagePasteEvent = {
+                    clipboardData: event.clipboardData,
+                    isImage: true,
+                    preventDefault: function () {
+                        this.defaultPrevented = true;
+                    },
+                    defaultPrevented: false
+                };
+
+                this.fireEvent( 'willPaste', imagePasteEvent);
```

Comment:
What if there is more than one image on the clipboard? Do we want to fire multiple paste events?